File: go/docmost/services/groups_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// GroupsAPI implements Groups API.
type GroupsAPI struct {
	http *httpclient.HTTPClient
}

func NewGroupsAPI(http *httpclient.HTTPClient) *GroupsAPI {
	return &GroupsAPI{http: http}
}

func (a *GroupsAPI) GetWorkspaceGroups(ctx context.Context, request *models.PaginationOptions) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "groups", request)
}

func (a *GroupsAPI) GetGroupInfo(ctx context.Context, request *models.GroupIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "groups/info", request)
}

func (a *GroupsAPI) CreateGroup(ctx context.Context, request *models.CreateGroupDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "groups/create", request)
}

func (a *GroupsAPI) UpdateGroup(ctx context.Context, request *models.UpdateGroupDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "groups/update", request)
}

func (a *GroupsAPI) GetGroupMembers(ctx context.Context) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "groups/members", nil)
}

func (a *GroupsAPI) AddGroupMembers(ctx context.Context, request *models.AddGroupUserDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "groups/members/add", request)
}

func (a *GroupsAPI) RemoveGroupMember(ctx context.Context, request *models.RemoveGroupUserDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "groups/members/remove", request)
}

func (a *GroupsAPI) DeleteGroup(ctx context.Context, request *models.GroupIdDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "groups/delete", request)
}

File: python/docmost/services/groups_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import AddGroupUserDto, CreateGroupDto, GroupIdDto, PaginationOptions, RemoveGroupUserDto, UpdateGroupDto


class GroupsApi:
    """Groups API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def get_workspace_groups(self, request: PaginationOptions) -> ApiResponse[Any]:
        return self._http.post("groups", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_group_info(self, request: GroupIdDto) -> ApiResponse[Any]:
        return self._http.post("groups/info", body=request.model_dump(by_alias=True, exclude_none=True))

    def create_group(self, request: CreateGroupDto) -> ApiResponse[Any]:
        return self._http.post("groups/create", body=request.model_dump(by_alias=True, exclude_none=True))

    def update_group(self, request: UpdateGroupDto) -> ApiResponse[Any]:
        return self._http.post("groups/update", body=request.model_dump(by_alias=True, exclude_none=True))

    def get_group_members(self) -> ApiResponse[Any]:
        return self._http.post("groups/members", body=None)

    def add_group_members(self, request: AddGroupUserDto) -> ApiResponse[Any]:
        return self._http.post("groups/members/add", body=request.model_dump(by_alias=True, exclude_none=True))

    def remove_group_member(self, request: RemoveGroupUserDto) -> ApiResponse[Any]:
        return self._http.post("groups/members/remove", body=request.model_dump(by_alias=True, exclude_none=True))

    def delete_group(self, request: GroupIdDto) -> ApiResponse[Any]:
        return self._http.post("groups/delete", body=request.model_dump(by_alias=True, exclude_none=True))

